boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -93,7 +93,17 @@ STRIP = $(CROSSDEV)strip --strip-unneeded
 AR = $(CROSSDEV)ar rcs
 endif
 NM = $(CROSSDEV)nm
-OBJCOPY = $(CROSSDEV)objcopy
+
+# Native compiler doesn't have objcopy ... just ignore it.
+ifeq ($(CONFIG_HOST_MACOS),y)
```

Comment:
```suggestion
ifneq ($(CONFIG_HOST_MACOS),y)
  OBJCOPY ?= $(CROSSDEV)objcopy
  OBJDUMP = $(CROSSDEV)objdump
endif
```

Reply:
That's not the same. On MACOS/ARM64 objcopy is not available and the call in arch/sim/src/Makefile is breaking. There are two potential solutions (except installing extra software on the host): Guard the call in arch/sim/src/Makefile or defining OBJCOPY as and empty call. The later one is currently used defining "OBJCOPY = :" in MACOS/ARM64. ':' is as empty call in bash/zsh shells.

Reply:
So, x64 MAC has objcopy/objdump by default?

Reply:
No, my x86_64 based Macbook running Big Sur also doesn't have objcopy while opjdump is available. This seems to be a clang based issue as both are using clang based host compiler suites. Updated this location.

Reply:
So, check CONFIG_HOST_MACOS is enough?

Reply:
Seems not ... current build breaks exactly there. Within the CI you're installaing gcc/objopy via brew. My solution is targetting to use the native compiler suite and not installing anything via brew for the simuation. This way the simulation would be easier to use for new community who wants to join. Just downloading the code is good enough work on NuttX with the simlation. I know, this could be philosophic discussion as the CI solution is harmonizing the simulation build across the host architectures. Anyway, the guard "ifeq ($(CONFIG_HOST_ARM64),y)" is needed.

Reply:
I also agree your idea to avoid the extra installation as much as possible. So I would suggest that it's better to left OBJCOPY/OBJDUMP empty on macOS regardless it run on arm64 or x64.

Reply:
There's no issue with objdump. It's part of native compiler suite. Only objcopy is missing. It needs to be guarded with ARM64. Otherwise the CI build for x86 simulation on MacOS fails as it's based on gcc compiler suite. If there's a need to harmonize the simulation on MacOS for X86 and M1 always using native compiler suite, that's an additional topic. It also needs involvement of the CI team. What's your view on this?

Reply:
It's better to align X86 and M1 CI setup on the same page, either gcc or clang is fine to me.
BTW, objcopy is useless on sim, because we can't use binary directly instead of elf. So, you can create a new patch to drop all objcopy reference from sim specifical directory.

Reply:
Not sure, if this is possible for all architectures and host OS versions. The gcc/binutils based versions seems to work different regarding fvisibility=hidden compared to clang based solution. At least on the MacOS host systems. Objcopy is used to hide the NuttX world from the HOST world avoiding calling wrong OS level functions. For my best knowledge all gcc/binutils based simulations needs to use objcopy.

Reply:
You are right. objcopy in sim is used to rename the symbols to avoid the conflict with host's glibc:
https://github.com/apache/incubator-nuttx/blob/master/arch/sim/src/Makefile#L288

Reply:
Ok, I fully understand the situation now. Look like the best approach is:

1. Change the default visibility to hidden for both macOS and Linux
2. Add ` __attribute ((visibility("default")))' Explicitly to the function need be called by host side code
3. Remove  nuttx-names.dat.in and the related hack
4. Remove OBJCOPY finally

Reply:
BTW, maybe you can create a new PR to fix this issue separately since it isn't related to M1 but the difference between Linux and macOS.

Reply:
Okay ... I get your point ... let me check, if I can manage that as it depends on having Linux/MacOS nut ARM64/X86_64 running. I assume, we need to validate at least 4 combination: MacOS/ARM64, MacOS/X86_64, Linux/X86_64, Linux/ARM64. Also it needs to be validated, that there's no regression on Windows/Cygwin based simulations. Do you know the level of support for BSD currently?

Reply:
You are right. Look like gcc and clang interpret hidden attribute for relocated file in a different way.